client/product-details/bnpl-site-messaging/style.scss

```diff
@@ -3,16 +3,32 @@
 	--wc-bnpl-margin-bottom: 0;
 }
 
-// Only apply the adjusted margins on the bnpl element when it immediately follows .price or .wp-block-woocommerce-product-price and the `ready` class has been added.
-.price,
-.wp-block-woocommerce-product-price {
-	&:has( + #payment-method-message.ready ) {
-		margin-bottom: 0;
+.pmme-loading {
+	animation: pmme-loading 1s linear infinite alternate;
+	background: #afafaf;
+}
+
+.summary,
+.wp-block-woocommerce-product-price + {
+	#payment-method-message {
+		height: 4rem;
+		position: relative;
+		overflow: hidden;
+
+		div,
+		iframe {
+			height: 100% !important;
+		}
 	}
 
-	+ #payment-method-message.ready {
-		margin-top: 0.5rem;
-		margin-bottom: var( --wc-bnpl-margin-bottom );
+	.pmme-loading {
+		position: absolute;
+		padding: 0;
+		margin: 0;
+		top: 0;
+		left: 0;
+		height: 100%;
+		width: 100%;
```

Comment:
These aren't really applying to the `.pmme-loading` element since it sits inside the `#payment-method-message` element. I also don't think these styles are necessarily needed since the container is being styled how you want it without them.

That being said, I _do_ think that this might look a bit better if we limit the width to 50%. This might make the skeleton better represent the content it's standing in for.

https://github.com/user-attachments/assets/73a67556-7b3e-40a0-a855-ddcde832d9ff

If you agree, this should be nested inside `#payment-method-message`, otherwise the entire `.pmme-loading` block can be deleted.
```
.pmme-loading {
    width: 50%;
}
```

Reply:
> These aren't really applying to the .pmme-loading element since it sits inside the #payment-method-message element. 

Nice catch! I actually refactored this to move the preloader inside `#payment-method-message` and forgot to move the CSS block accordingly. 

> I also don't think these styles are necessarily needed since the container is being styled how you want it without them.

I played around with these styles and I believe the `position: absolute` and `position: relative` on the parent (and height/width on the loader) are required at a minimum. Without fixing the preloader in place, when the PMME is ready and loads, for some reason removing and replacing the preloader causes the page to have a miniature little skip. See this example below...


https://github.com/user-attachments/assets/a3fe504a-d36a-4237-8316-40d828bd3a4c

Note that here without the `absolute` fixed positioning, right at the moment when the preloader is removed from the DOM, the page skips a tiny bit upwards.


https://github.com/user-attachments/assets/d708145c-b84b-41cf-970d-0b106208fe9a

Here with the `absolute` positioning, the PMME enters the scene slightly more smoothly, which was something rather critical I was trying to fix following this feedback -> https://github.com/Automattic/woocommerce-payments/issues/9126#issuecomment-2251654267

> That being said, I do think that this might look a bit better if we limit the width to 50%. This might make the skeleton better represent the content it's standing in for.

On very small screen widths, giving the preloader 50% width makes it too narrow and looks a little strange. 


https://github.com/user-attachments/assets/f004263a-e283-40cf-8098-cdba44b60b8a

See above where the loader is now smaller than the element it is loading.

I think I've managed to craft a compromise by keeping the width at 100%, but giving it a fixed max-width, so that even at its largest, the preloader will someone fit more closer to its contents.

All of the above changes added in 03caf4796e1b827bd99429c40ef2bb82c0cbf384. Let me know if those changes work for you. 🙏